backend/app/services/rag/pipeline/docling.py

```
@@ -0,0 +1,325 @@
+
+                logger.info(
+                    f"Docling conversion successful: {len(content)} bytes -> "
+                    f"{len(md_content)} characters"
+                )
+
+                return md_content
+
+        except httpx.TimeoutException:
+            raise DoclingServiceError(
+                f"Docling API request timed out after {self.timeout} seconds"
+            )
```

Ruff (0.15.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)